components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -2396,6 +2397,21 @@ private void parseSupportedGrantTypesConfig(OMElement oauthConfigElem) {
                         refreshTokenAllowedGrantTypes.put(grantTypeName, isRefreshAllowed);
                     }
                 }
+
+                // Read the public client allowed grant types from default grant types.
+                OMElement publicClientAllowedElement = supportedGrantTypeElement
+                        .getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.PUBLIC_CLIENT_ALLOWED));
+                String publicClientAllowed = null;
+                if (publicClientAllowedElement != null) {
+                    publicClientAllowed = publicClientAllowedElement.getText();
+                }
+                if (StringUtils.isNotEmpty(publicClientAllowed)) {
```

Comment:
The validation added here was reverted to the previous implementation with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2161/commits/0bdbde689d1f94f235e62337c7ca01c87a4a747d

Explanation:
Grant types added with isPublicClientAllowed property and value set to true will be added to `publicClientSupportedGrantTypes` list and value set to false will be added to `publicClientNotSupportedGrantTypes`. All default grant types will have the property set to either the value. 
In a scenario where the property is not mentioned in the custom grant type configuration, the grant type will not be added to either lists. So, if the custom grant type is added to the old array configuration, it will get added to the `publicClientSupportedGrantTypes` list

Reply:
Can u please add a comment to the code as well so that if someone tries to modify the code they could get the clear idea.

Reply:
Added with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2161/commits/64f406e714a253f713f4cb583ab708cb2a7725a6